R/zchunk_LA102.en_emiss_CDIAC.R

```diff
@@ -1,97 +1,155 @@
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+
+    # 2. Perform computations
+    # prep region IDs for matching
+    iso_GCAM_regID %>%
+      select(iso, GCAM_region_ID) ->
+      iso_GCAM_regID_reg
+
+    # append GCAM region IDs to isos and map GCAM fuels to CDIAC fuels
+    L100.CDIAC_CO2_ctry_hist %>%
+      gather(fuel, value, -iso, -year) %>%
+      left_join_error_no_match(iso_GCAM_regID_reg, by = "iso") %>%
+      rename(CDIAC_fuel = fuel) %>%
+      left_join_error_no_match(CDIAC_fuel, by = "CDIAC_fuel") %>%
+
+    # Aggregate CO2 emissions by GCAM region and fuel
+      group_by(GCAM_region_ID, fuel, year) %>%
+      summarise(value = sum(value) * CONV_KT_MT) ->
+      L102.CO2_Mt_R_F_Yh
+
+    # Calculate regional and global CO2 emissions coefficients by fuel
+    # Calculate the TPES by fuel, deducting non-energy use of fuels that does not result in CO2 emissions
+
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      filter(sector == "in_industry_feedstocks" & fuel %in% L102.CO2_Mt_R_F_Yh$fuel) %>%
+      left_join_error_no_match(A32.nonenergy_Cseq, by = c("fuel" = "subsector")) %>%
+      mutate(val = value * remove.fraction) %>%
+      select(GCAM_region_ID, fuel, year, val) ->
+      L102.en_sequestered_EJ_R_Fi_Yh
+
+    # subtracts the non-energy use of fuels (sequestered carbon) from the TPES to get only the emitting energy
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      filter(sector == "TPES" & fuel %in% L102.CO2_Mt_R_F_Yh$fuel) %>%
+      left_join_error_no_match(L102.en_sequestered_EJ_R_Fi_Yh, by = c("GCAM_region_ID", "fuel", "year")) %>%
+      mutate(val = value - val) %>%
+      select(-sector, -value) ->
+      L102.en_emitted_EJ_R_Fi_Yh
+
+    # Default emissions coefficients to replace NA values where below energy consumption is 0 (kgC/GJ)
+    DEFAULT_GAS_CCOEF <- 14.2
+    DEFAULT_COAL_CCOEF <- 27.3
+    DEFAULT_LIQUIDS_CCOEF <- 19.6
+
+    # Calculate the regional emissions coefficients by fuel, using only the energy whose carbon is assumed to be emitted
+    L102.CO2_Mt_R_F_Yh %>%
+      filter(fuel %in% L102.en_emitted_EJ_R_Fi_Yh$fuel) %>%
+      left_join_error_no_match(L102.en_emitted_EJ_R_Fi_Yh, by = c("GCAM_region_ID", "fuel", "year")) %>%
+      mutate(value = value / val) %>%
+      select(-val) %>%
+
+    # reset to defaults wherever NAs result from 0 energy consumption
+      mutate(value = if_else(fuel == "gas" & is.na(value), DEFAULT_GAS_CCOEF, value)) %>%
+      mutate(value = if_else(fuel == "coal" & is.na(value), DEFAULT_COAL_CCOEF, value)) %>%
+      mutate(value = if_else(fuel == "refined liquids" & is.na(value), DEFAULT_LIQUIDS_CCOEF, value)) ->
+      L102.Ccoef_kgCGJ_R_F_Yh
+
+    # aggregate regional values to global and calculate global coefficients
+    L102.en_emitted_EJ_R_Fi_Yh %>%
+      group_by(fuel, year) %>%
+      summarise(value = sum(val)) ->
+      L102.en_emitted_EJ_Fi_Yh
+
+    L102.CO2_Mt_R_F_Yh %>%
```

Comment:
a comment would be helpful to clarify value, value.x and value.y (global average coef, emissions, energy)